src/modules/layout/components/Header.js

```diff
@@ -171,9 +168,10 @@ class Header extends React.PureComponent<PropsType> {
           {cityModel && <HeaderText theme={theme}>{this.cityDisplayName(cityModel)}</HeaderText>}
         </HorizontalLeft>
         <MaterialHeaderButtons>
-          {this.renderItem('Search', 'search', 'always', !this.isPeeking() ? this.goToSearch : undefined)}
-          {this.renderItem('Change Language', 'language', 'always',
-            !this.isPeeking() && goToLanguageChange ? goToLanguageChange : undefined)}
+          {!peeking && categoriesAvailable &&
+            this.renderItem('Search', 'search', 'always', this.goToSearch)}
```

Comment:
Why do we hide the search, when categories are not available? Does that make the header glitch? (we should avoid glitches)

Reply:
Because search is a search for categories and it does not make sense to show it if categories are not available. If there are no categories, a loading spinner is displayed. This doesn't make sense if the fetching of categories failed (or they are still loading).

Reply:
Also, the header kind of glitches anyway, since only when the categories are pushed, the city name is displayed in the header. And I really think it looks good :) But test it yourself

Reply:
I just tested this and it looks pretty nice. The missing content in the header just plops into place. No glitching or something.